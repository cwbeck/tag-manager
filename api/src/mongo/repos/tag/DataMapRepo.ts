import { injectable } from 'inversify';
import DataMap from '../../models/tag/DataMap';
import UnderRevisionControl from './abstractions/UnderRevisionControl';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class DataMapRepo extends UnderRevisionControl<DataMap> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                ___persisting_id: 1,
                _revision_id: 1,
            },
            unique: true,
        },
    ];
}
