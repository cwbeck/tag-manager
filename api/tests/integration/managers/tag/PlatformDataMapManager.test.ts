import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import DataMapRepo from '../../../../src/mongo/repos/tag/DataMapRepo';
import PlatformEventRepo from '../../../../src/mongo/repos/tag/PlatformEventRepo';
import Mock = jest.Mock;

describe('PlatformDataMapManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a platform data map via event', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatformRevision();

        const mockEventModel = TestUtils.buildMockPlatformEvent();
        const mockEventRepo = {} as PlatformEventRepo;
        mockEventRepo.findByIdThrows = jest.fn().mockResolvedValue(mockEventModel);
        const mockSutModel = TestUtils.buildMockPlatformDataMap();
        const mockSutRepo = {} as DataMapRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        mockSutRepo.findByIds = jest.fn().mockResolvedValue([mockSutModel]);

        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockImplementation((model: any) => {
            if (model.name === 'PlatformEvent') {
                return mockEventRepo;
            }
            return mockSutRepo;
        });

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatformEvent(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                    platform_data_maps {
                        ...platformDatamapsFields
                    }
                }
            }
            fragment platformDatamapsFields on PlatformDataMap {
                id
                key
                var_type
                input_type
                description
                icon
                option_values
                validation_rules {
                    type
                    input_value
                }
                is_optional
                default_value {
                    __typename
                    ... on DefaultValueContainer {
                        value
                    }
                    ... on DefaultValueContainerArray {
                        values
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
