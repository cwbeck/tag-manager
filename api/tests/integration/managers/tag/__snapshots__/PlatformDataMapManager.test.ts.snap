// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformDataMapManager fetches a platform data map via event 1`] = `
Object {
  "data": Object {
    "getPlatformEvent": Object {
      "id": "5edf4c48e9615d9a1c66cddd",
      "name": "test platform event",
      "platform_data_maps": Array [
        Object {
          "default_value": null,
          "description": "platform data map description",
          "icon": null,
          "id": "5edf4c48e9615d9a1c66cdd0",
          "input_type": "TEXT",
          "is_optional": true,
          "key": "pdm",
          "option_values": null,
          "validation_rules": null,
          "var_type": "STRING",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
